src/main/java/rx/internal/operators/OperatorValve.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.operators;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import rx.Observable;
+import rx.Observable.Operator;
+import rx.Producer;
+import rx.Subscriber;
+
+/**
+ * An {@code Observable} that emits the first {@code num} items emitted by the source {@code Observable}.
+ * <p>
+ * <img width="640" height="305" src="https://raw.githubusercontent.com/wiki/ReactiveX/RxJava/images/rx-operators/take.png" alt="" />
+ * <p>
+ * You can choose to pay attention only to the first {@code num} items emitted by an {@code Observable} by using
+ * the {@code take} operator. This operator returns an {@code Observable} that will invoke a subscriber's
+ * {@link Subscriber#onNext onNext} function a maximum of {@code num} times before invoking
+ * {@link Subscriber#onCompleted onCompleted}.
+ */
+public final class OperatorValve<T> implements Operator<T, T> {
+    private final Observable<Boolean> onByDefault;
+    private final long granularity;
+
+    public OperatorValve(Observable<Boolean> onByDefault, long granularity) {
+        this.onByDefault = onByDefault;
+        this.granularity = granularity;
+    }
+
+    @Override
+    public Subscriber<? super T> call(final Subscriber<? super T> child) {
+        return new Subscriber<T>(child) {
+            private Producer p;
+            private long backlog = 0;// synchronized access on Producer p
```

Comment:
I don't know if it gets optimized away or not, but when one debugs the class, the single step will go over all of these field initializations. And for me, it's distracting and prompts if the author meant some reasonable initial value or not.